components/admin/product/productUpdate/productUpdate.tsx

```diff
@@ -70,22 +84,23 @@ export default function Dashboard({ index }: { index: number | undefined }) {
   const router = useRouter();
   const token = useToken((state) => state.token);
   const data = useStore((state) => (state.data));
-  const setData = useStore((state) => state.setData);
+
   useEffect(() => {
     if (index != undefined) {
+      console.log("fetchedData", fetchedData)
       console.log("index", index)
-      setRawdata(data[index])
+      { fetchedData && setRawdata(fetchedData[index]) }
     }
 
-  })
+  },[])
```

Comment:
Here fetchedData is missing in useEffect dependency array, so It wont work properly as expected